Gems/PhysX/Code/Editor/ColliderComponentMode.h

```diff
@@ -63,12 +65,9 @@ namespace PhysX
 
         AzToolsFramework::ViewportUi::ClusterId
             m_modeSelectionClusterId; //!< Viewport UI cluster for changing sub mode.
-        AzToolsFramework::ViewportUi::ButtonId
-            m_dimensionsModeButtonId; //!< Id of the Viewport UI button for resize/dimensions mode.
-        AzToolsFramework::ViewportUi::ButtonId
-            m_offsetModeButtonId; //!< Id of the Viewport UI button for offset mode.
-        AzToolsFramework::ViewportUi::ButtonId
-            m_rotationModeButtonId; //!< Id of the Viewport UI button for rotation mode.
+
+        AZStd::vector<AzToolsFramework::ViewportUi::ButtonId> m_buttonIds; //!< Ids for the Viewport UI buttons for each mode.
```

Comment:
For this case it might be more appropriate to use a fixed array `AZStd::array<ButtonId, SubMode::NumModes>`, rather than a vector, with size `SubMode::NumModes`. That way the size is what is expected from the enumeration.

Reply:
Good idea, updated to fixed size vector

Reply:
Actually I'm going to switch back to a vector, for the same reason I gave below - there might not always be a 1 to 1 mapping between buttons and modes.